backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/AbstractInterestComputingMethod.java

```diff
@@ -145,24 +141,16 @@ public ComputingResponse compute(final @NonNull ComputingRequest request)
 				.contractModuleId(request.getModularContractModuleId())
 				.lockOwner(request.getLockOwner())
 				.build();
-		final ModularContractLogEntriesList modularContractLogEntries = modularContractLogService.getModularContractLogEntries(query);
-		if (modularContractLogEntries.isEmpty())
+		final PInstanceId pInstanceId = modularContractLogService.getModularContractLogEntrySelection(query);
```

Comment:
super minor: IMHO u can call it `selectionId` instead `pInstanceId` (which sounds weird) :)